example/index.js

```diff
@@ -269,11 +279,70 @@ function onParamsChange() {
 
 	}
 
+	// Denoiser
+	if ( params.limitlessSamples ) pathTracer.maxSamples = Number.POSITIVE_INFINITY;
+	else pathTracer.maxSamples = params.maxSamples;
+	const renderScaleController = findController( 'renderScale' );
+	const oldEnableState = pathTracer.enableDenoiser;
+	pathTracer.enableDenoiser = params.enableDenoiser;
+	pathTracer.renderAux = params.renderAux;
+	pathTracer.denoiserDebugging = params.denoiserDebugging;
+	if ( oldEnableState !== params.enableDenoiser ) {
+
+		// force the renderscale to be 1 when using the denoiser
+		pathTracer.renderScale = 1;
+		params.renderScale = 1;
+		// set and disable the renderScale gui
+		if ( renderScaleController ) {
+
+			renderScaleController.disable();
+			renderScaleController.setValue( 1 );
+
+		}
```

Comment:
When the pixeled are scaled OIDN rejects them. Theres documentation on it somewhere but it considers them as surface microstructures and treats it as intentional. Each pixel has to be 1 pixel in the WebGLTexture. 

If we can somehow get access to the base (unscaled) texture we can denoise that and then scale the result. Otherwise using renderScale will break the denoiser.

Reply:
See https://github.com/gkjohnson/three-gpu-pathtracer/pull/663#discussion_r1689741838